src/main/java/net/pms/dlna/DLNAImageProfile.java

```diff
@@ -0,0 +1,341 @@
+		}
+	}
+
+	/**
+	 * Converts the {@link String} passed as argument to a
+	 * {@link DLNAImageProfile}. If the conversion fails, this method returns
+	 * {@code null}.
+	 */
+	public static DLNAImageProfile toDLNAImageProfile(String argument) {
+		return toDLNAImageProfile(argument, null);
+	}
+
+	/**
+	 * Converts the integer passed as argument to a {@link DLNAImageProfile}.
+	 * If the conversion fails, this method returns {@code null}.
+	 */
+	public static DLNAImageProfile toDLNAImageProfile(int value) {
+		return toDLNAImageProfile(value, null);
+	}
+
+	/**
+	 * Converts the integer passed as argument to a {@link DLNAImageProfile}. If the
+	 * conversion fails, this method returns the specified default.
+	 */
+	public static DLNAImageProfile toDLNAImageProfile(int value, DLNAImageProfile defaultImageProfile) {
+		switch (value) {
+			case GIF_LRG_INT:
+				return GIF_LRG;
+			case JPEG_LRG_INT:
+				return JPEG_LRG;
+			case JPEG_MED_INT:
+				return JPEG_MED;
+			case JPEG_RES_H_V_INT:
+				return JPEG_RES_H_V;
+			case JPEG_SM_INT:
+				return JPEG_SM;
+			case JPEG_TN_INT:
+				return JPEG_TN;
+			case PNG_LRG_INT:
+				return PNG_LRG;
+			case PNG_TN_INT:
+				return PNG_TN;
+			default:
+				return defaultImageProfile;
+		}
+	}
+
+	/**
+	 * Converts the {@link String} passed as argument to a
+	 * {@link DLNAImageProfile}. If the conversion fails, this method returns
+	 * the specified default.
+	 */
+	public static DLNAImageProfile toDLNAImageProfile(String argument, DLNAImageProfile defaultImageProfile) {
+		if (argument == null) {
+			return defaultImageProfile;
+		}
+
+		argument = argument.toUpperCase(Locale.ROOT);
+		switch (argument) {
+			case "GIF_LRG":
+				return DLNAImageProfile.GIF_LRG;
+			case "JPEG_LRG":
+				return DLNAImageProfile.JPEG_LRG;
+			case "JPEG_MED":
+				return DLNAImageProfile.JPEG_MED;
+			case "JPEG_RES_H_V":
+				return DLNAImageProfile.JPEG_RES_H_V;
+			case "JPEG_SM":
+				return DLNAImageProfile.JPEG_SM;
+			case "JPEG_TN":
+				return DLNAImageProfile.JPEG_TN;
+			case "PNG_LRG":
+				return DLNAImageProfile.PNG_LRG;
+			case "PNG_TN":
+				return DLNAImageProfile.PNG_TN;
+			default:
+				if (argument.startsWith("JPEG_RES")) {
+					Matcher matcher = Pattern.compile("^JPEG_RES_?(\\d+)[X_](\\d+)").matcher(argument);
+					if (matcher.find()) {
+						return new DLNAImageProfile(
+							JPEG_RES_H_V_INT,
+							"JPEG_RES_" + matcher.group(1) + "_" + matcher.group(2),
+							Integer.parseInt(matcher.group(1)),
+							Integer.parseInt(matcher.group(2))
+						);
+					}
+				}
+				return defaultImageProfile;
+		}
+	}
+
+	public String getMimeType() {
+		switch (imageProfileInt) {
+			case GIF_LRG_INT:
+				return "image/gif";
+			case JPEG_LRG_INT:
+			case JPEG_MED_INT:
+			case JPEG_RES_H_V_INT:
+			case JPEG_SM_INT:
+			case JPEG_TN_INT:
+				return "image/jpeg";
+			case PNG_LRG_INT:
+			case PNG_TN_INT:
+				return "image/png";
+		}
+		return "image/unknown";
```

Comment:
@Sami32 I just "made that up" as it's never supposed to happen. All possible values are listed above, and the only possibility to code will ever get there is if a new profile is added but this ```switch``` isn't updated. Ideally it should fail if that happened, so that it would be easily detected. 

I didn't put anything there at first, but the Java compiler requires there to be a return value for "all potential paths" so I got an error without it. That's why I added it, thinking that it'll never be used. Thinking about it now, it might be better to throw an exception if the code ever gets there, so that it's easier to pick up and fill in the correct mime type for the profile.